# Bio_Analysis
In this project, we have 3 type of datasets: __RNA-seq__, __scRNA-seq__ and __scATAC-seq__.

## RNA-seq
In this part, there are two scripts: __RNA-seq.R__ and __Diff.R__  
In the script in __RNA-seq.R__, you can see that  how I analyse the data for __GO enrichment__ and __different expression genes__.  
In the script in __Diff.R__, you can see that how I compare the result of __Gfold__ and __DESeq2__.

## scRNA-seq
In this part, you can see that __Seurat__ can analyse the data for __clustering__, __finding marker genes__ and __inferring the trajectory of development__.

## scATAC-seq
In this part, there are two scripts: __scATAC-seq.R__ and __CHIPseeker.R__  
In the script in __scATAC-seq.R__, you can see that __SnapATAC__ can analyse the data for __clustering__, __peak calling__ and __annotating___.  
In the script in __CHIPseeker.R__, you can see the __annotating for peak calling__, __analysing the proportion of gene types__.

## Reference

### Article
> 1. Bolger A M, Lohse M, Usadel B. Trimmomatic: a flexible trimmer for Illumina sequence data[J]. Bioinformatics, 2014, 30(15): 2114-2120.
> 2. Fang R, Preissl S, Li Y, et al. Comprehensive analysis of single cell ATAC-seq data with SnapATAC[J]. Nature communications, 2021, 12(1): 1-15.
> 3. Love M I, Huber W, Anders S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2[J]. Genome biology, 2014, 15(12): 1-21.
> 4. Liao Y, Smyth G K, Shi W. The Subread aligner: fast, accurate and scalable read mapping by seed-and-vote[J]. Nucleic acids research, 2013, 41(10): e108-e108.
> 5. Li H, Handsaker B, Wysoker A, et al. The sequence alignment/map format and SAMtools[J]. Bioinformatics, 2009, 25(16): 2078-2079.
> 6. Li H, Durbin R. Fast and accurate short read alignment with Burrows-Wheeler transform[J]. Bioinformatics, 2009; 25: 1754â€“60.
> 7. Qiu X, Mao Q, Tang Y, et al. Reversed graph embedding resolves complex single-cell trajectories[J]. Nature methods, 2017, 14(10): 979-982.
> 8. Satija R, Farrell J A, Gennert D, et al. Spatial reconstruction of single-cell gene expression data[J]. Nature biotechnology, 2015, 33(5): 495-502.
> 9. Yu G, Wang L G, He Q Y. ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization[J]. Bioinformatics, 2015, 31(14): 2382-2383.

### Website
split the data for scRNA-seq: <https://cloud.tencent.com/developer/article/1606139>  
split the data for scATAC-seq: <https://www.jianshu.com/p/ab98dd332368>  
build the index for scRNA-seq: <https://www.jianshu.com/p/ed9f7f9ad203>  
why the datasize become smaller after sorting: <https://www.jianshu.com/p/6ed1bfbb7b72>  
analyse the RNA-seq: <https://www.jianshu.com/p/80c7bf8a2834>  
the protocol for Seurat: <https://satijalab.org/seurat/articles/pbmc3k_tutorial.html>  
the protocol for SnapATAC: <https://github.com/r3fang/SnapTools>  
the protocol for monocle3: <https://github.com/cole-trapnell-lab/monocle3/blob/master/examples/website_script.R>  
ananotate the peak: <https://www.jianshu.com/p/c76e83e6fa57>
